daemon/container.go

```diff
@@ -1250,9 +1253,15 @@ func (container *Container) updateParentsHosts() error {
 
 		if c != nil && !container.daemon.config.DisableNetwork && container.hostConfig.NetworkMode.IsPrivate() {
 			logrus.Debugf("Update /etc/hosts of %s for alias %s with ip %s", c.ID, ref.Name, container.NetworkSettings.IPAddress)
-			if err := etchosts.Update(c.HostsPath, container.NetworkSettings.IPAddress, ref.Name); err != nil {
+			if err := etchosts.UpdateIPv4(c.HostsPath, container.NetworkSettings.IPAddress, ref.Name); err != nil {
 				logrus.Errorf("Failed to update /etc/hosts in parent container %s for alias %s: %v", c.ID, ref.Name, err)
 			}
+			if container.NetworkSettings.GlobalIPv6Address != "" {
```

Comment:
(Just thinking out loud); if the container could provide a list of its IP-addresses (IPv4 and IPv6), then all IP-addresses could be passed to `etchosts.Update()` (with a slightly modified signature), which could simply write `hostname + strings.join(IPs, ' ')` (fake code) after validating that all IPs are either valid v4 or v6 addresses.

This could reduce the duplication between the two UpdateIPvXXX functions and (I think) solve the problem with duplicate entries for the same hostname in `/etc/hosts`